packages/server/src/AccountsServer.js

```diff
@@ -205,6 +208,84 @@ export class AccountsServer {
   }
 
   /**
+   * @description Sets an impersonation rule.
+   * @param {string} userId - User id.
+   * @param {Function<boolean>} usersFilterFn - impersonation predicate.
+   * @returns {boolean} - true if user can impersonate to impersonation user.
+   */
+  setImpersonationRule(userId: string, usersFilterFn: Function<any, boolean>) {
```

Comment:
This behaviour is not flexible enough. It means that at the stage of my server startup, I need to know which userIds are able to impersonate other users. Or rather I need to configure that every time my "list" of my privileged users changes. A much more flexible and easier approach is to allow a simple function of the following signature `(user: UserObject, impersonateToUsername: string) => Promise<boolean>` and this will potentially do whatever is needed to come up with the answer